src/components/dev/button/button.component.tsx

```diff
@@ -16,12 +18,26 @@ export interface DevButtonProps {
 export function DevButton({
   label,
   to,
-  onClick,
+  onClick: onClickProp,
   variant = "primary",
   isLoading,
   isDisabled
 }: DevButtonProps) {
-  // TODO: Throw an error if both `to` and `onClick` are set, or add code to handle that properly.
+  const { navigate } = useLocation();
+
+  const onClick = useMemo(() => {
```

Comment:
@Jonathansoufer If you have integrated the routing functionality into your `Button` or any other clickable component, you might want to also transfer that change so that a button with `onClick` and `to` waits for the `onClick` to resolve before navigating. Otherwise, this can cause issues in some screens/actions like downloading a recovery file (because if you navigate before the wallet is marked as backed up, you'll be redirected back to the same screen).